api/rest-client.ts

```diff
@@ -285,29 +285,27 @@ export class RingRestClient {
           return this.request(options)
         }
         logError(
-          `http request failed.  ${url} returned unknown errors: (${stringify(
-            errors
-          )}).`
+          `http request failed.  ${url} returned unknown errors: (${stringify(errors)}).`
         )
       }
 
-      if (response.status === 404 && url.startsWith(clientApiBaseUrl)) {
+      if (response.statusCode === 404 && url.startsWith(clientApiBaseUrl)) {
         logError('404 from endpoint ' + url)
-        if (response.data?.error?.includes(hardwareId)) {
+        if (response.body?.error?.includes(hardwareId)) {
```

Comment:
I'm not sure if this is the correct logic anymore, but something I'll have to test on my end

Reply:
When I was debugging that part, e.reponse was actually set. (It is thrown from the **got** method in **requestWithRetry**, where the try/catch just forwards the error object when it contains the e.response property (__throw e__ in catch). If not, another error is being logged and a 5s-delayed retry is set up.